File: INTRODUCCION-POO/ejercicio2/bus.py
# Clase Bus que modela la abstracción de un bus
class Bus:
	def __init__(self, capacidad):
		self.capacidad = capacidad
		self.pasajeros = 0
		self.caja = 0.0

	def subir_pasajeros(self, cantidad):
		disponibles = self.capacidad - self.pasajeros
		if cantidad <= disponibles:
			self.pasajeros += cantidad
			print(f"Subieron {cantidad} pasajeros.")
		else:
			self.pasajeros = self.capacidad
			print(f"Solo subieron {disponibles} pasajeros, el bus está lleno.")

	def cobrar_pasaje(self):
		costo = 1.50
		total = self.pasajeros * costo
		self.caja += total
		print(f"Se cobraron Bs. {total:.2f} a {self.pasajeros} pasajeros.")
		self.pasajeros = 0  # Todos pagan y bajan

	def asientos_disponibles(self):
		disponibles = self.capacidad - self.pasajeros
		print(f"Asientos disponibles: {disponibles}")
		return disponibles

# programa principal

bus = Bus(capacidad=30)
bus.subir_pasajeros(25)
bus.asientos_disponibles()
bus.cobrar_pasaje()
bus.asientos_disponibles()
bus.subir_pasajeros(10)
bus.asientos_disponibles()

File: INTRODUCCION-POO/ejercicio2/Bus.java
package ejercicio2;
// Ejercicio 2. Realiza la abstracción de un Bus.
// a) Al bus desean subir X cantidad de pasajeros, actualiza los datos del bus.
// b) Crea un método para cobrar pasaje a los pasajeros. (Costo del pasaje: bs. 1.50)’
// c) Muestra cuántos asientos quedan disponibles.
// d) Crea una instancia del bus y utiliza los métodos de los incisos anteriores.
public class Bus {
    //atributos
    private int capacidad;
    private int pasajerosActuales;
    private double costoPasaje = 1.50;

    //constructor
    public Bus(int capacidad) {
        this.capacidad = capacidad;
        this.pasajerosActuales = 0; //el bus esta vacio        
    }

    //metodos
    public void subirPasajero(int numPasajeros){
        if(this.pasajerosActuales + numPasajeros <= this.capacidad){
            this.pasajerosActuales += numPasajeros;
            System.out.println(numPasajeros + " pasajeros subiendo al bus.");

        }
        else{
            System.out.println("No hay espacio para " + numPasajeros + " pasajeros.");
        }        
    }

    public double cobrarPasaje(){
        return this.pasajerosActuales * this.costoPasaje;
    }

    public int getAsientosDisponibles(){
        return this.capacidad - this.pasajerosActuales;
    }

    public void mostrarBus(){
        System.out.println("Capacidad del bus: " + this.capacidad);
        System.out.println("Pasajeros actuales: " + this.pasajerosActuales);
        System.out.println("Asientos disponibles:" + getAsientosDisponibles());
        System.out.println("Total recaudado: Bs. " + cobrarPasaje());
    }

}
